### ISSUE TITLE
Plugin incompatible with gradle 2.2

### ISSUE THREAD

**Issue by dmitry-treskunov** [etiennestuder/teamcity-build-scan-plugin#18]
What is the minimum version of Gradle this plugin is compatible with? For Gradle 2.2.1 (I know, it's pretty old) it doesn't work with the following error. Is it possible to support old versions of Gradle?

```
Gradle failure report
15:35:59
    FAILURE: Build failed with an exception.
15:35:59
    
15:35:59
    * Where:
15:35:59
    Initialization script 'C:\BuildAgent\temp\agentTmp\build-scan-init.gradle' line: 34
15:35:59
    
15:35:59
    * What went wrong:
15:35:59
    Could not find property 'pluginManager' on settings 'ExportToMavenTool'.
15:35:59
    
15:35:59
    * Try:
15:35:59
    Run with --info or --debug option to get more log output.
15:35:59
    
15:35:59
    * Exception is:
15:35:59
    groovy.lang.MissingPropertyException: Could not find property 'pluginManager' on settings 'ExportToMavenTool'.
15:35:59
      at org.gradle.api.internal.AbstractDynamicObject.propertyMissingException(AbstractDynamicObject.java:43)
15:35:59
      at org.gradle.api.internal.AbstractDynamicObject.getProperty(AbstractDynamicObject.java:35)
15:35:59
      at org.gradle.api.internal.CompositeDynamicObject.getProperty(CompositeDynamicObject.java:94)
15:35:59
      at org.gradle.initialization.DefaultSettings_Decorated.getProperty(Unknown Source)
15:35:59
      at build_scan_init_94swg0je7djxu5eai5echpff5$_run_closure3.doCall(C:\BuildAgent\temp\agentTmp\build-scan-init.gradle:34)
15:35:59
      at org.gradle.listener.ClosureBackedMethodInvocationDispatch.dispatch(ClosureBackedMethodInvocationDispatch.java:40)
15:35:59
      at org.gradle.listener.ClosureBackedMethodInvocationDispatch.dispatch(ClosureBackedMethodInvocationDispatch.java:25)
15:35:59
      at org.gradle.listener.BroadcastDispatch.dispatch(BroadcastDispatch.java:83)
15:35:59
      at org.gradle.listener.BroadcastDispatch.dispatch(BroadcastDispatch.java:31)
15:35:59
      at org.gradle.messaging.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
15:35:59
      at com.sun.proxy.$Proxy11.settingsEvaluated(Unknown Source)
15:35:59
      at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:122)
15:35:59
      at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:106)
15:35:59
      at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:86)
15:35:59
      at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:80)
15:35:59
      at org.gradle.launcher.cli.ExecuteBuildAction.run(ExecuteBuildAction.java:33)
15:35:59
      at org.gradle.launcher.cli.ExecuteBuildAction.run(ExecuteBuildAction.java:24)
15:35:59
      at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:36)
15:35:59
      at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:26)
```

**Comment by etiennestuder** [comment]
Gradle 4.1. I do not intend to support any older versions.

**Comment by etiennestuder** [comment]
Just for the record, Gradle 2.2.1 was released on Nov 24, 2014. For a project I maintain in my spare time, I'm not willing to have this kind of backward compatibility.

**Comment by dmitry-treskunov** [comment]
Hi, I understand that there is no need to support old versions of Gradle.

But unfortunately the current implementation makes it impossible to install this plugin to any TeamCity Server that has at least one legacy project that uses an old version of Gradle. 
It's so because currently the plugin modifies **every** Gradle step with `build-scan-init.gradle`: even if this step uses an unsupported version of Gradle.

I'm not an expert in Gradle, but maybe it's possible to check the version of Gradle inside of the build-scan-init.gradle and do nothing if it's lower than 4.1. What do you think?

**Comment by etiennestuder** [comment]
Ah. Good idea. I will look into that. Thanks for the proposal.

**Comment by etiennestuder** [comment]
@dmitry-treskunov I've released [v0.17](https://github.com/etiennestuder/teamcity-build-scan-plugin/releases/tag/v0.17) that should fix the reported issue. The init script will bail out gracefully if the Gradle version used is <4.1.

The release can be found in the TC plugin portal and also on Bintray.






Thanks again for reporting this.
